### PR TITLE
[KYUUBI #1565] Move time functions to RowSetUtils and move schema to spark package

### PR DESCRIPTION
**Pull request by hddong**
### _Why are the changes needed?_

1. move time functions from Rowset to RowSetUtils
2. move schema to spark package

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/schema/RowSetSuite.scala [hunk 1]
```diff
@@ -237,7 +237,7 @@ class RowSetSuite extends KyuubiFunSuite {
     assert(r6.get(9).getStringVal.getValue === "2018-11-06")
 
     val r7 = iter.next().getColVals
-    assert(r7.get(10).getStringVal.getValue === "2018-11-17 13:33:33.600")
+    assert(r7.get(10).getStringVal.getValue === "2018-11-17 13:33:33.6")
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/schema/RowSetSuite.scala:240]: why change precision?

Comment (hddong) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/schema/RowSetSuite.scala:240]: > why change precision?

https://github.com/apache/incubator-kyuubi/blob/6fa1327f4e58ca79e7c07f8e089d6ca312b323c9/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/schema/RowSet.scala#L222-L223

The length of fraction is from 0 to 9, precision not changed, just the following '0' is not displayed.

Comment (yaooqinn) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/schema/RowSetSuite.scala:240]: We shall keep the padding

Comment (hddong) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/schema/RowSetSuite.scala]: increased the minWidth

File: kyuubi-common/src/main/scala/org/apache/kyuubi/util/RowSetUtils.scala [hunk 2]
```diff
@@ -18,11 +18,35 @@
...
 import java.nio.ByteBuffer
+import java.time.chrono.IsoChronology
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeFormatterBuilder
+import java.time.temporal.ChronoField
+import java.util.Locale
 
 import scala.language.implicitConversions
 
 object RowSetUtils {
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/util/RowSetUtils.scala:29]: let's add ` private[kyuubi]` here

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
LGTM

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**